client/my-sites/controller.js

```diff
@@ -334,22 +338,23 @@ module.exports = {
 		} else {
 			// if sites has fresh data and siteId is invalid
 			// redirect to allSitesPath
-			if ( ! isRequestingSites( getState() ) ) {
+			if ( ! isRequestingSites( state ) ) {
 				return page.redirect( allSitesPath );
 			}
 
 			let waitingNotice;
 			let freshSiteId;
 			const selectOnSitesChange = () => {
 				// if sites have loaded, but siteId is invalid, redirect to allSitesPath
-				freshSiteId = getSiteId( getState(), siteFragment );
+				const freshState = getState();
```

Comment:
👍 on naming this "freshState" and for caching the state for use in the next several calls.